features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Listing/ApiTableRow.js

```diff
@@ -16,63 +16,223 @@
+        if (status === 404) {
+          this.setState({ notFound: true });
+        }
+      });
+  }
+
+  render() {
+    let details_link = "/apis/" + this.props.api.id;
+    const { name, version, context, description } = this.props.api;
+    if (!this.state.active) {
+      // Controls the delete state, We set the state to inactive on delete success call
+      return null;
     }
+    const { classes } = this.props;
+    const apiRows = [
+      <TableRow
+        hover
+        onClick={this.handleClick}
+        key={"row-data-" + this.props.api.id}
+      >
+        <TableCell className={this.state.collapseOpen ? classes.root : null}>
+          <Link to={details_link}>{name}</Link>
+        </TableCell>
+        <TableCell className={this.state.collapseOpen ? classes.root : null}>
+          {version}
+        </TableCell>
+        <TableCell className={this.state.collapseOpen ? classes.root : null}>
+          {context}
+        </TableCell>
+        <TableCell className={this.state.collapseOpen ? classes.root : null}>
+          <StarRatingBar rating={this.state.rating} />
+        </TableCell>
+      </TableRow>
+    ];
+
+    // If the state of collapsible row is open, render the row with additional details of the api
+    if (this.state.collapseOpen) {
+      apiRows.push(
+        <TableRow key={"row-data-child-" + this.props.api.id}>
+          <TableCell colSpan="4">
+            <Typography variant="body2">{this.state.description}</Typography>
+            <Typography variant="body2">
+              Policies:
+              {this.state.policies.length === 0 ? (
+                <span>&lt; NOT SET FOR THIS API &gt;</span>
+              ) : null}
+              {this.state.policies.map((policy, index) => (
+                <span key={"span-policy-" + index}>
+                  {" " + policy}
+                  {this.state.policies.length !== 1 &&
+                    index !== this.state.policies.length - 1 && <span>,</span>}
+                </span>
+              ))}
+            </Typography>
+            <Typography variant="body2">
+              Subscribed Apps:
+              {this.state.subscribedApplications.length === 0 ? (
+                <span>&lt; NOT SET FOR THIS API &gt;</span>
+              ) : null}
+              {this.state.subscribedApplications.map((app, index) => (
+                <Link
+                  to={"/applications/" + app.value}
+                  key={"app-link-" + app.value}
+                >
+                  {" " + app.label}
+                  {this.state.subscribedApplications.length !== 1 &&
+                    index !== this.state.subscribedApplications.length - 1 && (
+                      <span>,</span>
+                    )}
+                </Link>
+              ))}
+            </Typography>
+          </TableCell>
+        </TableRow>
+      );
+    }
+    return apiRows;
+  }
+}
+
+class Star extends React.Component {
+  constructor(props) {
```

Comment:
IMO this constructor is not needed here and even we could make it ( `Star` ) a function component

Reply:
Removed the constructor and made Star as a function component.